### DISCUSSION TITLE
Promiscuous mode

### DISCUSSION THREAD

**Discussion by islameehassan** [GyulyVGC/sniffnet#469] [Q&A] [answered]
If Sniffnet's main aim is to monitor traffic meant for my PC, why would it configure the NIC in promiscuous mode? Does not this raise some legal concerns about it?

**Comment by GyulyVGC** [comment]
Promiscuous mode is set by default on all the major packet sniffers, including Wireshark.
This is considered necessary in this kind of tools to enable all-encompassing network monitoring and administration activities.

For what concerns law, using packet sniffers on public networks (e.g., your job network) is generally discouraged, and it's up to the users to make proper use of it.

**Reply by islameehassan** [reply]
So it would not be necessary to keep this mode on if I use Sniffnet for bandwidth monitoring for my PC.

**Reply by GyulyVGC** [reply, marked as answer]
In most of the cases it doesn't make any difference from your PC perspective, so having it ON or OFF isn't a big deal as long as you don't have particular legal constraints.

**Reply by youssef-mansor** [reply]
so how can I disable the mode given that I would like to only view traffic meant for my PC, i. e. what files should I edit in the source code?

**Reply by GyulyVGC** [reply]
Hi @youssef-mansor, that's super easy, you just need to set false instead of true at [this line](https://github.com/GyulyVGC/sniffnet/blob/main/src/networking/types/capture_context.rs#L15).

**Reply by youssef-mansor** [reply]
Thank you :)  however after disabling the mode as you mentioned, I built the project again and it seems to still shows connections that neither has the source nor the destination of my PC.

**Reply by GyulyVGC** [reply]
Mm, that's weird.
Are you sure those are not multicast / broadcast addresses?

**Comment by GyulyVGC** [comment]
Hey @islameehassan @youssef-mansor @AlyamanMas

I've found out you are maintaining a Sniffnet's fork with some interesting features implemented (e.g., process identification).
I still have to compile it but from the commit messages the work you were doing seems promising.

Would you be interested in becoming official contributors to the project? 
There are some items in my roadmap that could greatly benefit from your work, for example one the most requested features is #170.

**Reply by islameehassan** [reply]
Hey @GyulyVGC, it is really nice hearing from you. I would really love to contribute to the project of course.

**Reply by GyulyVGC** [reply]
I tried it on my macOS from https://github.com/AlyamanMas/sniffnet but it fails.
Maybe you developed it just for Linux?
Anyway, if you with your friends would like to submit such functionality, PRs are welcomed.
I'd like to support at least Windows, macOS, and Linux though.
